src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -18,11 +18,12 @@ internal void Initialize(CircuitClientProxy clientProxy)
 
         protected override void BeginInvokeJS(long asyncHandle, string identifier, string argsJson)
         {
-            if (_clientProxy == CircuitClientProxy.OfflineClient)
+            if (!_clientProxy.Connected)
             {
-                var errorMessage = "JavaScript interop calls cannot be issued while the client is not connected, because the server is not able to interop with the browser at this time. " +
-                    "Components must wrap any JavaScript interop calls in conditional logic to ensure those interop calls are not attempted during periods where the client is not connected.";
-                throw new InvalidOperationException(errorMessage);
+                throw new InvalidOperationException("JavaScript interop calls cannot be issued at this time. This is because the component is being " +
```

Comment:
Do you think it might be worth having a custom exception type for this? I could imagine developers wanting the most robust code would ultimately have no choice but to try/catch it if their goal is to skip a given JSInterop call if disconnected (because even if they just checked `componentContext.IsConnected`, the connection state could change before their next line of code executes).

Reply:
I think so. I'll file a bug. This would need to go in .Components though